lib/rubocop/cop/style/hash_syntax.rb

```diff
@@ -96,7 +96,21 @@ def word_symbol_pair?(pair)
           return false unless key.sym_type?
 
           sym_name = key.loc.expression.source
-          sym_name !~ /\A:["']|=\z/
+          return false if ends_with_equals? sym_name
+          valid_19_syntax_symbol? sym_name
+        end
+
+        def valid_19_syntax_symbol?(sym_name)
```

Comment:
We used to have a similar regexp, but I recently removed it to fix some other problem. Have to go through the history, but I definitely don't want to employ a regexp for this.